src/core/thread/mle_router.cpp

```diff
@@ -2943,6 +2948,28 @@ ThreadError MleRouter::GetChildInfoByIndex(uint8_t aChildIndex, otChildInfo &aCh
     return error;
 }
 
+ThreadError MleRouter::AddChild(const otChildInfo &aChildInfo)
+{
+    ThreadError error = kThreadError_None;
+    Child *child;
+
+    VerifyOrExit((child = NewChild()) != NULL, error = kThreadError_NoBufs);
+
+    memset(child, 0, sizeof(*child));
+
+    memcpy(&child->mMacAddr, &aChildInfo.mExtAddress, sizeof(child->mMacAddr));
+    child->mValid.mRloc16 = aChildInfo.mRloc16;
+    child->mTimeout = aChildInfo.mTimeout;
+    child->mMode = aChildInfo.mRxOnWhenIdle | aChildInfo.mSecureDataRequest | aChildInfo.mFullFunction |
```

Comment:
`mRxOnWhenIdle`, `mSecureDataRequest`, etc. are booleans.  You need to translate that into the proper bitfield for `mMode`.